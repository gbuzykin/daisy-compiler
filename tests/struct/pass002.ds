namespace N1 {

struct A1 {
  i : i32;
  f : double;
};

namespace N2 {

struct A2 {
  i : i32;
  f : double;
};

func f1(x : A1) {}
func f2(x : A2) {}
func f3(x : N1::A1) {}
func f4(x : N2::A2) {}
func f5(x : ::N1::A1) {}
func f6(x : ::N1::N2::A2) {}

}

func f7(x : A1) {}
func f8(x : N2::A2) {}

}

func f9(x : N1::A1) {}
func f10(x : N1::N2::A2) {}
func f11(x : ::N1::A1) {}
func f12(x : ::N1::N2::A2) {}
