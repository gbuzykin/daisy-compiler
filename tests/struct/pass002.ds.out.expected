./struct/pass002.ds:4:3: debug: defining field `i`
 4 |   i : i32;
   |   ^
./struct/pass002.ds:5:3: debug: defining field `f`
 5 |   f : double;
   |   ^
./struct/pass002.ds:11:3: debug: defining field `i`
 11 |   i : i32;
    |   ^
./struct/pass002.ds:12:3: debug: defining field `f`
 12 |   f : double;
    |   ^
./struct/pass002.ds:15:6: debug: defining function `f1`
 15 | func f1 (x : A1) {}
    |      ^~
./struct/pass002.ds:15:1: debug: function `f1` definition
 15 | func f1 (x : A1) {}
    | ^~~~~~~~~~~~~~~~
./struct/pass002.ds:16:6: debug: defining function `f2`
 16 | func f2 (x : A2) {}
    |      ^~
./struct/pass002.ds:16:1: debug: function `f2` definition
 16 | func f2 (x : A2) {}
    | ^~~~~~~~~~~~~~~~
./struct/pass002.ds:17:6: debug: defining function `f3`
 17 | func f3 (x : N1::A1) {}
    |      ^~
./struct/pass002.ds:17:1: debug: function `f3` definition
 17 | func f3 (x : N1::A1) {}
    | ^~~~~~~~~~~~~~~~~~~~
./struct/pass002.ds:18:6: debug: defining function `f4`
 18 | func f4 (x : N2::A2) {}
    |      ^~
./struct/pass002.ds:18:1: debug: function `f4` definition
 18 | func f4 (x : N2::A2) {}
    | ^~~~~~~~~~~~~~~~~~~~
./struct/pass002.ds:19:6: debug: defining function `f5`
 19 | func f5 (x : ::N1::A1) {}
    |      ^~
./struct/pass002.ds:19:1: debug: function `f5` definition
 19 | func f5 (x : ::N1::A1) {}
    | ^~~~~~~~~~~~~~~~~~~~~~
./struct/pass002.ds:20:6: debug: defining function `f6`
 20 | func f6 (x : ::N1::N2::A2) {}
    |      ^~
./struct/pass002.ds:20:1: debug: function `f6` definition
 20 | func f6 (x : ::N1::N2::A2) {}
    | ^~~~~~~~~~~~~~~~~~~~~~~~~~
./struct/pass002.ds:24:6: debug: defining function `f7`
 24 | func f7 (x : A1) {}
    |      ^~
./struct/pass002.ds:24:1: debug: function `f7` definition
 24 | func f7 (x : A1) {}
    | ^~~~~~~~~~~~~~~~
./struct/pass002.ds:25:6: debug: defining function `f8`
 25 | func f8 (x : N2::A2) {}
    |      ^~
./struct/pass002.ds:25:1: debug: function `f8` definition
 25 | func f8 (x : N2::A2) {}
    | ^~~~~~~~~~~~~~~~~~~~
./struct/pass002.ds:29:6: debug: defining function `f9`
 29 | func f9 (x : N1::A1) {}
    |      ^~
./struct/pass002.ds:29:1: debug: function `f9` definition
 29 | func f9 (x : N1::A1) {}
    | ^~~~~~~~~~~~~~~~~~~~
./struct/pass002.ds:30:6: debug: defining function `f10`
 30 | func f10 (x : N1::N2::A2) {}
    |      ^~~
./struct/pass002.ds:30:1: debug: function `f10` definition
 30 | func f10 (x : N1::N2::A2) {}
    | ^~~~~~~~~~~~~~~~~~~~~~~~~
./struct/pass002.ds:31:6: debug: defining function `f11`
 31 | func f11 (x : ::N1::A1) {}
    |      ^~~
./struct/pass002.ds:31:1: debug: function `f11` definition
 31 | func f11 (x : ::N1::A1) {}
    | ^~~~~~~~~~~~~~~~~~~~~~~
./struct/pass002.ds:32:6: debug: defining function `f12`
 32 | func f12 (x : ::N1::N2::A2) {}
    |      ^~~
./struct/pass002.ds:32:1: debug: function `f12` definition
 32 | func f12 (x : ::N1::N2::A2) {}
    | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
./struct/pass002.ds: info: warnings 0, errors 0
